### PR TITLE
OCM-2428: Remove openshift-v prefix from create cluster version attribute

### PR DESCRIPTION
**Pull request by nirarg**

### CODE REVIEW

File: provider/cluster_rosa_classic_resource.go [hunk 1]
```diff
@@ -686,7 +686,7 @@ func createClassicClusterObject(ctx context.Context,
 			return nil, errors.New(errHeadline + "\n" + description)
 		}
 		vBuilder := cmv1.NewVersion()
-		versionID := state.Version.Value
+		versionID := fmt.Sprintf("openshift-v%s", state.Version.Value)
```
Comment (bardielle) [inline provider/cluster_rosa_classic_resource.go:689]: I think better moving the format string to a const

File: provider/cluster_rosa_classic_resource.go [hunk 2]
```diff
@@ -1321,7 +1318,7 @@ func (r *ClusterRosaClassicResource) upgradeClusterIfNeeded(ctx context.Context,
 	}
 
 	// Make sure the desired version is available
-	versionId := state.CurrentVersion.Value
+	versionId := fmt.Sprintf("openshift-v%s", state.CurrentVersion.Value)
```
Comment (bardielle) [inline provider/cluster_rosa_classic_resource.go:1321]: Use the same const as I mentioned in my previous comment

File: subsystem/cluster_resource_rosa_test.go [hunk 3]
```diff
@@ -2563,7 +2563,7 @@ var _ = Describe("ocm_cluster_rosa_classic - upgrade", func() {
 					worker_role_arn = "",
 				}
 			}
-			version = "openshift-v4.10.1"
+			version = "4.10.1"
```
Comment (bardielle) [inline subsystem/cluster_resource_rosa_test.go:2566]: Can you add a failure test that uses an incorrect version string?

File: provider/cluster_rosa_classic_resource.go (hunk 1 above)
Comment (nirarg) [reply provider/cluster_rosa_classic_resource.go:689]: Please notice that here removing the prefix from `state.Version.Value` and in the next comment removing it from `CurrentVersion.Value`
I don't think const is required here

File: subsystem/cluster_resource_rosa_test.go (hunk 3 above)
Comment (nirarg) [reply subsystem/cluster_resource_rosa_test.go:2708]: Please see new comment added https://github.com/terraform-redhat/terraform-provider-ocm/pull/124/commits/530b0881efb093ebfe052f6fd001164cba419894

**Review by bardielle — APPROVED** [review]

**Pull request merged**